clients/fides-js/rollup.config.mjs

```diff
@@ -0,0 +1,90 @@
+import copy from "rollup-plugin-copy";
+import dts from "rollup-plugin-dts";
+import esbuild from "rollup-plugin-esbuild";
+import filesize from "rollup-plugin-filesize";
+import nodeResolve from "@rollup/plugin-node-resolve";
+
+const name = "fides";
+const isDev = process.env.NODE_ENV === "development";
+const GZIP_SIZE_ERROR_KB = 20; // fail build if bundle size exceeds this
+const GZIP_SIZE_WARN_KB = 15; // log a warning if bundle size exceeds this
+
+/**
+ * @type {import('rollup').RollupOptions}
+ */
+export default [
+  {
+    input: `src/${name}.ts`,
+    plugins: [
+      nodeResolve(),
+      esbuild({
+        minify: !isDev,
+      }),
+      copy({
+        // Automatically add the built script to the privacy center's static files for bundling:
+        targets: [{ src: `dist/${name}.js`, dest: "../privacy-center/public/lib/" }],
+        verbose: true,
+        hook: "writeBundle",
+      }),
+      filesize({
+        reporter: [
+          "boxen", // default reporter, which prints a nice CLI output
+
+          // Add a defensive check to fail the build if our bundle size starts getting too big!
+          (options, bundle, { bundleSize, gzipSize, fileName }) => {
+            const gzipSizeKb = Number(gzipSize.replace(" KB", ""));
+            if (gzipSizeKb > GZIP_SIZE_ERROR_KB) {
+              console.error(
+                `❌ ERROR: ${fileName} build failed! Gzipped size (${gzipSize}) exceeded maximum size (${GZIP_SIZE_ERROR_KB} KB)!`,
+                `If you must, update GZIP_SIZE_* constants in clients/fides-js/rollup.config.mjs`,
+              );
+              process.exit(1);
+            } else if (gzipSizeKb > GZIP_SIZE_WARN_KB) {
+              console.warn(
+                `️🚨 WARN: ${fileName} build is getting large! Gzipped size (${gzipSize}) exceeded warning size (${GZIP_SIZE_WARN_KB} KB)!`,
+                `If you must, update GZIP_SIZE_* constants in clients/fides-js/rollup.config.mjs`,
+              );
+              if (isDev) {
+                process.exit(1);
+              }
+            } else {
+              console.log(
+                `✅ ${fileName} gzipped size passed maximum size checks (${gzipSize} < ${GZIP_SIZE_ERROR_KB} KB)`
+              );
+            }
+          },
```

Comment:
FYI: I just wired this up to keep our eye on bundle size. I set a max target of 20 KB (compressed), which is still pretty big, but didn't want to overdo it yet.

Reply:
This is great! I feel way better having bundle size checks automated  🚀